framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -2448,8 +2465,120 @@ protected boolean isTypeCastSafe(AnnotatedTypeMirror castType, AnnotatedTypeMirr
             }
         }
 
+        // Check when casting the expression having type T (a type variable) to type T,
+        // as compiler will not report the unchecked warning in this case (so we should),
+        // and we cannot statically verify the subtype relation of the annotations.
+        // See CastFromTtoT.java:20 for an example.
```

Comment:
I don't quite understand this part..

Reply:
I have refined the comment and the example. Hope now it's more clear